tests/eas/load_tracking.py

```diff
@@ -537,6 +540,206 @@ def test_load_avg_behaviour(self, experiment, tasks):
+        """
+        Compute the expected utilization per CPU given a phase.
+        :param tasks: json description of the tasks
+        :type tasks: dict
+
+        :param cpus: list of CPUs for which the expected utilization needs to
+                     be computed
+        :type cpus: [int]
+
+        :param phase: name of the phase for which the expected utilization
+                      needs to be computed
+        :type phase: str
+        :return a list containing the expected utilization for each CPU
+        """
+        expected = []
+        for cpu in cpus:
+            expected_val = 0
+            for task in tasks.iteritems():
+                period_us = task[1]['phases'][phase]['timer']['period']
+                run_time_us = task[1]['phases'][phase]['run']
+                duty_cycle = run_time_us * 100 / period_us
+                [phase_cpu] = task[1]['phases'][phase]['cpus']
+                expected_val += 1024 * (duty_cycle / 100.0) \
+                                if phase_cpu == cpu else 0
+            expected.append(expected_val)
+
+        return expected
+
+    def _get_cpus(self, tasks):
+        """
+        Return the list of CPUs where the different tasks are executed
+        :param tasks: json description of the tasks
+        :type tasks: dict
+        :return a list of CPUs on which tasks are running
+        """
+        cpus = []
+        phases = tasks[tasks.keys()[0]]['phases'].keys()
+        for phase in phases:
+            for task in tasks.keys():
+                [cpu] = tasks[task]['phases'][phase]['cpus']
+                if cpu not in cpus:
+                    cpus.append(cpu)
+
+        return cpus
+
+    def _get_stable_range(self, df, tasks, phase_id):
+        """
+        Return a start and end interval for which the utilization signal should
+        be stabilized.
+        :param df: data frame of the sched_switch events
+        :type df: DataFrame
+
+        :param tasks: json description of the tasks
+        :type tasks: dict
+
+        :param phase_id: index (starting from 0) of the phase for which the
+                         stable range is computed
+        :type phase: int
+        :return a start time and a end time where the utilization signal should
+                be stabilized.
+        """
+        task_names = tasks.keys()
+        phase_duration_s = 0
+        phase_names = tasks[task_names[0]]['phases'].keys()
+        df = df[df.next_comm.isin(task_names)]
+        start = df.index[1]
+        # Compute the total duration of the previous phases to get the start
+        # time of the given phase
+        for phase in (phase_names[: phase_id+1]):
+            start += phase_duration_s
+            loop = tasks[task_names[0]]['phases'][phase]['loop']
+            period_us = tasks[task_names[0]]['phases'][phase]['timer']['period']
+            phase_duration_s = (loop * period_us) / 1e6
+
+        end = start + phase_duration_s - UTIL_AVG_CONVERGENCE_TIME
```

Comment:
Why do we trim at the end?... the signal should be stable at that time... and by double trimming we need not necessary longer phases, isn't it?

Reply:
Agree. Done, I removed the trimming for the end value.